command/agent/cache/crypto/k8s.go

```diff
@@ -0,0 +1,97 @@
+package crypto
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping"
+	"github.com/hashicorp/go-kms-wrapping/wrappers/aead"
+)
+
+var _ KeyManager = (*KubeEncryptionKey)(nil)
+
+type KubeEncryptionKey struct {
+	renewable bool
+	wrapper   *aead.Wrapper
+}
+
+// NewK8s returns a new instance of the Kube encryption key. Kubernetes
+// encryption keys aren't renewable.
+func NewK8s(existingKey []byte) (*KubeEncryptionKey, error) {
```

Comment:
Nit: The constructor's naming should match the returned struct name (i.e. `NewKubeEncryptionKey` in this case).